sources_test/path.cpp

```diff
@@ -0,0 +1,163 @@
+#include "utility/file_system/path.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(PathCombine, test_combine_file_3_words)
+{
+
+    std::string path = util::Path::combine<std::string_view>(
+        false, "user", "Desktop", "kus.png");
+
+    EXPECT_STREQ(path.c_str(), "user/Desktop/kus.png");
+}
+TEST(PathCombine, test_combine_folder_3_words)
+{
+    std::string path =
+        util::Path::combine<std::string_view>(true, "user", "Desktop", "kus");
+
+    EXPECT_STREQ(path.c_str(), "user/Desktop/kus/");
+}
+
+TEST(PathCombine, test_combine_file_1_word)
+{
+
+    std::string path =
+        util::Path::combine<std::string_view>(false, "us!!dD  Der.png");
+
+    EXPECT_STREQ(path.c_str(), "us!!dD  Der.png");
+}
+TEST(PathCombine, test_combine_folder_1_word)
+{
+    std::string path =
+        util::Path::combine<std::string_view>(true, "useR  WQr/");
+
+    EXPECT_STREQ(path.c_str(), "useR  WQr/");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Please use
//------------------------------------------------------------------------------

Also we need to comment logical blocks, for example:
//------------------------------------------------------------------------------
// Simple test for paths
//------------------------------------------------------------------------------
<tests>
//------------------------------------------------------------------------------
// Difficult tests
//------------------------------------------------------------------------------
<tests>
//------------------------------------------------------------------------------